ios/MaasAccount/Sources/MaasAccount/Requirements/Components/ToggleView.swift

```diff
@@ -0,0 +1,86 @@
+import SwiftUI
+import Swappable
+
+public struct ToggleView<ToggledContent: View, UntoggledContent: View>: View  {
+
+    @Binding var isOn: Bool
+
+    private let toggledContent: ToggledContent
+    private let untoggledContent: UntoggledContent
+
+    public init(
+        isOn: Binding<Bool>,
+        @ViewBuilder toggledContent: () -> ToggledContent,
+        @ViewBuilder untoggledContent: () -> UntoggledContent
+    ) {
+        self._isOn = isOn
+        self.toggledContent = toggledContent()
+        self.untoggledContent = untoggledContent()
+    }
+
+    public var body: some View {
+        isOn
+            ? toggledContent.erased
+            : untoggledContent.erased
+    }
+}
+
+// MARK: - Checkbox
+
+public struct Checkbox: View, Swappable {
+
+    public struct InputType {
+        public let isOn: Binding<Bool>
+        public let toggledContent: AnyView?
+        public let untoggledContent: AnyView?
+    }
+    public var input: InputType
+
+    // TODO: - Constants
+    let width: CGFloat = 24
+    let height: CGFloat = 24
+    @Environment(\.colorPrimary) var primaryColor
+
+    init(isOn: Binding<Bool>) {
```

Comment:
not sure about the color.. probably would like the ability to override color in `init`, or pass `.foregroundColor` from outside - then no color should be defined in this component

Reply:
This will be passed via constants or provide it with `init`.

Reply:
Just noticed, that we'll use an Asset 🤦 so i'm removing `.primaryColor`.